scipy/stats/_rvs_sampling.py

```diff
@@ -168,3 +170,258 @@ def rvs_ratio_uniforms(pdf, umax, vmin, vmax, size=1, c=0, random_state=None):
+    >>> time_once(time_me)
+    '11.9222 ms'  # may vary
+
+    After generating the fast numerical inverse, subsequent calls to its
+    methods are much faster.
+    >>> time_once(lambda: fni.ppf(p))
+    '0.0819 ms'  # may vary
+
+    The fast numerical inverse can also be used to generate random variates
+    using inverse transform sampling.
+
+    >>> time_once(lambda: fni.rvs(size=100))
+    '0.0911 ms'  # may vary
+
+    Depending on the implementation of the distribution's random sampling
+    method, the random variates generated may be nearly identical, given
+    the same random state.
+
+    >>> rvs1 = dist.rvs(size=100, random_state=0)
+    >>> rvs2 = fni.rvs(size=100, random_state=0)
+    >>> np.allclose(rvs1, rvs2)
+    True
+
+    """
+    def __init__(self, dist, tol=1e-12, max_intervals=100000):
+        H, eu, intervals = _fast_numerical_inverse(dist, tol, max_intervals)
+        self.H = H
+        self.midpoint_error = eu
+        self.intervals = intervals
+
+    def ppf(self, q):
+        r'''
+        Approximate percent point function (inverse of `cdf`) at q of the
+        given RV.
+
+        Parameters
+        ----------
+        q : array_like
+            lower tail probability
+
+        Returns
+        -------
+        x : array_like
+            quantile corresponding to the lower tail probability q.
+
+        '''
+        q = np.asarray(q)  # no harm; self.H always returns an array
+        result = np.zeros_like(q, dtype=np.float64)
+        i = (q >= 0) & (q <= 1)
+        result[i] = self.H(q[i])
+        result[~i] = np.nan
+        return result
+
+    def rvs(self, size=1, random_state=None, qmc_engine=None):
+        """
+        Random variates of the given RV.
+
+        Parameters
+        ----------
+        size : int or tuple of ints, optional
+            Defining number of random variates (Default is 1).
+        random_state : (see below), optional
+            Defines the object to use for drawing pseudorandom variates.
+            If `random_state` is `None` the `~np.random.RandomState` singleton
+            is used.
+            If `random_state` is an int, a new ``RandomState`` instance is
+            used, seeded with `random_state`.
+            If `random_state` is already a ``RandomState`` or ``Generator``
+            instance, then that object is used.
+            Default is None.
+        qmc_engine : scipy.stats.qmc.QMCEngine(d=1), optional
+            This parameter defines the object to use for drawing
+            quasi-Monte Carlo variates. Default is None.
+        Returns
+        -------
+        rvs : ndarray or scalar
+            Random variates of given `size`.
+        """
+        if qmc_engine is not None and random_state is not None:
+            message = "Only one of `random_state` or `qmc_engine` may be used."
+            raise ValueError(message)
+        rng = qmc_engine or check_random_state(random_state)
+        return self.ppf(rng.random(size))
```

Comment:
```suggestion
        return self.ppf(rng.random(size))
```
Probably should use `uniform` for `random_state` and `random` for `qmc` to avoid problems with old NumPy.